packages/flutter/lib/src/widgets/spell_check.dart

```diff
@@ -328,3 +352,38 @@ void _addComposingRegionTextSpans(
     )
   );
 }
+
+/// Methods for showing spell check suggestions for misspelled words in toolbar:
+
+/// Finds specified [SuggestionSpan] that matches the provided index using
+/// binary search.
+SuggestionSpan? findSuggestionSpanAtCursorIndex(
```

Comment:
I've gotten pushback in code review before for adding a global function like this.  Is there any class that it would make sense to put this in?  As a static method.